File: frontend/src/types/macro-events.ts
export type MacroEventImpact = "low" | "medium" | "high";

export interface MacroEvent {
  event_id: string;
  name: string;
  name_ja: string | null;
  country: string;
  scheduled_at: string;
  impact: MacroEventImpact;
  estimate: string | null;
  actual: string | null;
  previous: string | null;
  source: string;
}

export interface MacroEventsResponse {
  events: MacroEvent[];
  source: string;
  window_days: number;
  fetched_at: string | null;
  note_ja: string | null;
}

File: backend/app/schemas/macro_events.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

MacroEventImpact = Literal["low", "medium", "high"]


class MacroEvent(BaseModel):
    event_id: str
    name: str
    name_ja: str | None = None
    country: str
    scheduled_at: datetime
    impact: MacroEventImpact
    estimate: str | None = None
    actual: str | None = None
    previous: str | None = None
    source: str = "finnhub"


class MacroEventsResponse(BaseModel):
    events: list[MacroEvent] = Field(default_factory=list)
    source: str = "finnhub"
    window_days: int = 7
    fetched_at: datetime | None = None
    note_ja: str | None = None
